lib/improver/utilities/solar.py

```diff
@@ -0,0 +1,277 @@
+            longitudes needs to be between 180.0 and -180.0
+        day_of_year (int):
+            Day of the year 0 to 365, 0 = 1st January
+        utc_hour (float):
+            Hour of the day in UTC in hours
+
+    Returns:
+        solar_elevation (numpy.array):
+            Solar elevation in degrees
+    """
+    if np.min(longitudes) < -180.0 or np.max(longitudes) > 180.0:
+        msg = ('Longitudes must be between -180.0 and 180.0')
+        raise ValueError(msg)
+    if np.min(latitudes) < -90.0 or np.max(latitudes) > 90.0:
+        msg = ('Latitudes must be between -90.0 and 90.0')
+        raise ValueError(msg)
+    declination = solar_declination(day_of_year)
+    decl = np.radians(declination)
+    hour_angle = solar_hour_angle(longitudes, day_of_year, utc_hour)
+    rad_hours = np.radians(hour_angle)
+    lats = np.radians(latitudes)
+    # Calculate solar position:
+
+    solar_elevation = (np.arcsin(np.sin(decl) * np.sin(lats) +
+                                 np.cos(decl) * np.cos(lats) *
+                                 np.cos(rad_hours)))
+
+    solar_elevation = np.degrees(solar_elevation)
+
+    return solar_elevation
+
+
+def daynight_terminator(longitudes, day_of_year, utc_hour):
+    """
+    Calculate the Latitude values of the daynight terminator
+
+    Args:
+        longitudes (numpy.array):
+            Array of longitudes.
+            longitudes needs to be between 180.0 and -180.0 degrees
+        day_of_year (int):
+            Day of the year
+        utc_hour (float):
+            Hour of the day in UTC
+
+    Returns:
+        latitudes (numpy.array):
+            latitudes of the daynight terminator
+    """
+    declination = solar_declination(day_of_year)
+    decl = np.radians(declination)
+    hour_angle = solar_hour_angle(longitudes, day_of_year, utc_hour)
+    rad_hour = np.radians(hour_angle)
+    lats = np.arctan(-np.cos(rad_hour)/np.tan(decl))
+    lats = np.degrees(lats)
+    return lats
+
+
+class DayNightMask(object):
+    """
+    Generate a daynight mask for the provided cube
+    """
+    def __init__(self):
+        """ Initial the DayNightMask Object """
+        self.NIGHT = 0
+        self.DAY = 1
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<DayNightMask : '
+                  'Day = {}, Night = {}>'.format(self.DAY, self.NIGHT))
+        return result
+
+    def _create_daynight_mask(self, cube):
+        """
+        Create blank daynight mask cube
+
+        Args:
+            cube (iris.cube.Cube):
+                cube with the times and coordinates required for mask
+
+        Returns:
+            daynight_mask (iris.cube.Cube):
+                Blank daynight mask cube. The resulting cube will be the
+                same shape as the time, y, and x coordinate, other coordinates
+                will be ignored although they might appear as attributes
```

Comment:
Can we clear up any unwanted attributes explicitly?  This would be tidier from a metadata perspective.

Reply:
Not easily as we wouldn't know what to try excluding. I would not expect the daynight_mask to be used externally but if it is saved we should look at this again but probably best to do this at the point you are saving the mask or alternatively when creating the mask only provided a cube of time, x and y rather than any old cube.